node-rfc/run_rfc_task.js

```diff
@@ -0,0 +1,209 @@
+"use strict";
+
+var rfc = require("node-rfc");
+var fs = require("fs");
+
+module.exports = function(grunt) {
+
+    // Project specific variables
+    var abapDevelopmentUser = process.env.ABAP_DEVELOPMENT_USER;
+    var abapDevelopmentPassword = process.env.ABAP_DEVELOPMENT_PASSWORD;
+    var abapDevelopmentServer = process.env.ABAP_DEVELOPMENT_SERVER;
+    var abapDevelopmentInstance = process.env.ABAP_DEVELOPMENT_INSTANCE;
+    var abapDevelopmentClient = process.env.ABAP_DEVELOPMENT_CLIENT;
+    var abapApplicationName = process.env.ABAP_APPLICATION_NAME;
+    var abapApplicationDesc = process.env.ABAP_APPLICATION_DESC;
+    var abapPackage = process.env.ABAP_PACKAGE;
+    var zipFileURL = process.env.ZIP_FILE_URL;
+    var gitCommit = process.env.GIT_COMMIT;
+    var targetDir = process.env.SAPDATADIR;
+
+    // Global Variables
+    var ctsDataFile = targetDir + "/CTS_Data.txt";
+
+    // Project configuration.
+    var abapConn = {
+        user: abapDevelopmentUser,
+        passwd: abapDevelopmentPassword,
+        ashost: abapDevelopmentServer,
+        sysnr: abapDevelopmentInstance,
+        client: abapDevelopmentClient
+    };
+    grunt.initConfig({
+        pkg: grunt.file.readJSON("package.json"),
+        createTransportRequest: {
+            options: {
+                conn: abapConn,
+                author: abapDevelopmentUser,
+                description: "Commit: " + gitCommit
```

Comment:
Seems to be weird to create a descrption here. Would be better IMO to provide the description from outside.
Two reasons:
1.) It might happen that a transport request receives artifacts from more than one commit.
2.) more "in general" when providing a generic description variable in the API the caller can still decide for the git commit in case it makes sense in that use case, but there is the freedom to provide also other information which might be important in a certain use case.